Ts api guardian

alexeagle opened the pull request:
/cc @alan-agius4

alexeagle committed:
test(bazel): Build and test ts-api-guardian locally

Also use it to test the public API for core and common

Once we have an ng_package for every package, we can remove
the npm dependency on ts-api-guardian and the gulp-based
public api check.

mhevery commented on modules/ts-api-guardian/lib/cli.ts:
```diff
@@ -1,4 +1,6 @@
-import chalk from 'chalk';
+// TODO(alexeagle): why not import chalk from 'chalk'?
+// Something to do with TS default export in UMD emit...
+const chalk = require('chalk');
```
so this does not work: `import * as chalk from 'chalk';`

alexeagle replied:
correct, doesn't work, I kinda remember why and could fix this todo before merging this.

mhevery commented on tools/public_api_guard/BUILD.bazel:
```diff
@@ -0,0 +1,26 @@
+
+# ts_api_guardian_test(
+#     name = "core_api",
+#     actual = "packages/core/npm_package/core.d.ts",
```
why commented out?

mary-poppins commented:
You can preview 9a657c9 at https://pr22544-9a657c9.ngbuilds.io/.

mhevery commented on modules/ts-api-guardian/lib/cli.ts:
```diff
@@ -75,6 +75,11 @@ export function startCli() {
+        // Under bazel, give instructions how to use bazel run to accept the golden file.
+        if (!!process.env['BAZEL_TARGET']) {
+          console.error('\n\nAccept the new golden file:');
+          console.error(`  bazel run ${process.env['BAZEL_TARGET']}.accept`);
```
this is inconsistent with `//tools/symbol-extractor:js_expected_symbol_test` since it uses .accept terminology. Could I convince you to update  `//tools/symbol-extractor:js_expected_symbol_test` to also have .accept?

mhevery commented:
Very cool!

mary-poppins commented:
You can preview 51af793 at https://pr22544-51af793.ngbuilds.io/.

mary-poppins commented:
You can preview e974071 at https://pr22544-e974071.ngbuilds.io/.

mary-poppins commented:
You can preview 51601bd at https://pr22544-51601bd.ngbuilds.io/.

mary-poppins commented:
You can preview 5a7bbc7 at https://pr22544-5a7bbc7.ngbuilds.io/.

mary-poppins commented:
You can preview 8c9853b at https://pr22544-8c9853b.ngbuilds.io/.

mary-poppins commented:
You can preview da976a6 at https://pr22544-da976a6.ngbuilds.io/.

mary-poppins commented:
You can preview 69cf33e at https://pr22544-69cf33e.ngbuilds.io/.

alan-agius4 commented:
This is awesome :)

On Fri, 02 Mar 2018 at 17:28, Mary Poppins <notifications@github.com> wrote:

> You can preview 69cf33e
> <https://github.com/angular/angular/commit/69cf33e66db318b036d920f3ec0011f048aa03e2>
> at https://pr22544-69cf33e.ngbuilds.io/.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/angular/angular/pull/22544#issuecomment-369972684>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AQv-WpxZazRdBYGKeLoTLPEJprAHGDpmks5taXMfgaJpZM4SZOTj>
> .
>

mary-poppins commented:
You can preview 441a7c4 at https://pr22544-441a7c4.ngbuilds.io/.

mary-poppins commented:
You can preview 21a3ba1 at https://pr22544-21a3ba1.ngbuilds.io/.

alexeagle committed:
build: copy ts-api-guardian sources

This is an exact mirror of https://github.com/angular/ts-api-guardian/tree/750f651eca4d57a4a7ebcc12c19d0beea1a62202
